ObservatoryHerald/AzureSpeechManager.cs

```diff
@@ -68,34 +45,63 @@ internal VoiceAzureCacheManager(HeraldSettings settings, HttpClient httpClient)
             }
 
             speech = new(speechConfig, null);
+
+            settings.Voices = PopulateVoiceSettingOptions();
         }
 
-        internal string GetAudioFileFromSsml(string ssml, string voice)
+        private Dictionary<string, object> PopulateVoiceSettingOptions()
         {
-            ssml = AddVoiceToSsml(ssml, voice);
-            string ssmlHash = ssml.GetHashCode().ToString("X");
+            ReadOnlyCollection<VoiceInfo> voices;
+
+            try
+            {
+                voices = speech.GetVoicesAsync().Result.Voices;
+            }
+            catch (Exception ex)
+            {
+                throw new PluginException("Herald", "Unable to retrieve voice list from Azure.", ex);
+            }
+            
+            var voiceOptions = new Dictionary<string, object>();
 
-            string audioFile;
 
-            if (cacheIndex.ContainsKey(ssmlHash))
+            foreach (var voice in voices)
             {
-                audioFile = cacheIndex[ssmlHash];
+                voiceOptions.Add(
+                    $"{voice.Locale} - {voice.LocalName}", 
```

Comment:
Do these voices change often at all? (Point is, if the selected voice LocalName changes, the selected voice will break when ObsCore attempts to use it on Azure.)

I'm fairly convinced this is good enough, but if we get reports of voices breaking, this could be an unstable value and we may need to consider a more stable ID to use.

Reply:
They do, but it's very infrequent, and Microsoft notifies in advance.